products/BellHybrid/apps/application-bell-settings/ApplicationBellSettings.cpp

```diff
@@ -176,15 +174,19 @@ namespace app
                                                                             std::move(prewakeUpLightDurationModel),
                                                                             std::move(prewakeUpFrontlightModel));
 
-                auto provider         = std::make_unique<bell_settings::PrewakeUpListItemProvider>(
-                    *prewakeUpSettingsModel, soundsRepository->getSongTitles());
+                const auto &preWakeupSoundsPath = paths::audio::proprietary() / paths::audio::preWakeup();
+                const auto &pathsSortingVector = std::vector<SoundsRepository::PathSorting>{
+                    {preWakeupSoundsPath, SoundsRepository::SortingBy::TitleAscending}};
+
+                auto soundsRepository = std::make_unique<SoundsRepository>(this, pathsSortingVector);
```

Comment:
you can also use a simplified constructor `SoundsRepository(app::ApplicationCommon *application, const PathSorting &path);`

Reply:
Oh, you're right! I just copied it from another place and didn't check there's a constructor better suiting the needs :wink: Fixed!